frontend/www/js/omegaup/components/Countdown.vue

```diff
@@ -6,37 +6,44 @@
 
 <script lang="ts">
 import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
+import { omegaup } from '../omegaup';
 import * as time from '../time';
 import * as ui from '../ui';
 import T from '../lang';
 
 @Component
 export default class Countdown extends Vue {
   @Prop() targetTime!: Date;
-  @Prop({ default: false }) countdownToNextSubmission!: boolean;
+  @Prop({
+    default: omegaup.CountdownFormat.EVENT_COUNTDOWN,
+  })
+  countdownFormat!: omegaup.CountdownFormat;
 
   timerInterval = 0;
   currentTime = Date.now();
 
   get timeLeft(): number {
-    if (!this.countdownToNextSubmission) {
-      return this.targetTime.getTime() - this.currentTime;
-    }
-    return Math.ceil((this.targetTime.getTime() - this.currentTime) / 1000);
+    return this.targetTime.getTime() - this.currentTime;
   }
 
   get formattedTimeLeft(): string {
-    if (!this.countdownToNextSubmission) {
+    if (this.countdownFormat === omegaup.CountdownFormat.EVENT_COUNTDOWN) {
       return time.formatDelta(this.timeLeft);
     }
-    return ui.formatString(T.arenaRunSubmitWaitBetweenUploads, {
-      submissionGap: this.timeLeft,
-    });
+    if (
+      this.countdownFormat ===
+      omegaup.CountdownFormat.WAIT_BETWEEN_UPLOADS_SECONDS
+    ) {
+      return ui.formatString(T.arenaRunSubmitWaitBetweenUploads, {
+        submissionGap: Math.ceil(this.timeLeft / 1000),
+      });
+    }
+    return '';
   }
 
   @Watch('timeLeft')
   onValueChanged(newValue: number): void {
-    if (newValue > 0) return;
+    if (newValue / 1000 > 0) return;
```

Comment:
```suggestion
    if (newValue > 0) return;
```

esta expresión es equivalente (se pueden multiplicar ambos lados de la desigualdad por la misma constante, en este caso 1000) y es un poco más rápida.

Reply:
Listo.